### PR TITLE
Android implementation

### PR DESCRIPTION
**Pull request by fAns1k**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "react-native-twitter-sdk",
+  "version": "0.0.1",
+  "description": "Twitter sdk for react-native",
+  "main": "src/index.js",
+  "scripts": {
+    "ci": "scripts/local-ci.sh",
+    "test:ios": "xcodebuild -project ios/TPSStore.xcodeproj/ -scheme 'TPSStoreTests' -sdk iphonesimulator -destination 'platform=iOS Simulator,name=iPhone 6' test",
```
Comment (itsmepetrov) [inline package.json]: I think we don't have unit tests yet

File: example/index.android.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
+/**
+ * Sample React Native App
+ * https://github.com/facebook/react-native
+ * @flow
+ */
```
Comment (itsmepetrov) [inline example/index.android.js]: Remove unnecessary comments

File: example/index.android.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import React, { Component } from 'react';
+import {
+  AppRegistry,
+  StyleSheet,
+  Text,
+  View
+} from 'react-native';
+import { TwitterLoginButton } from 'react-native-twitter-sdk'
+
+export default class example extends Component {
```
Comment (itsmepetrov) [inline example/index.android.js]: Please move this component to `src` folder

File: src/TwitterLoginButton.android.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+     } else if (eventDict.type === 'logoutFinished') {
+       if (this.props.onLogoutFinished) {
+         this.props.onLogoutFinished();
+       }
+     }
+   }
+
+   render() {
+     return (
+       <RCTTwitterLoginButton
```
Comment (isnifer) [inline src/TwitterLoginButton.android.js]: Maybe we need to use TPS namespace instead of RCT?

**Comment by isnifer** [comment]
I have successfully authenticated, but...
not_authenticated

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+
+'use strict';
```
Comment (isnifer) [inline src/index.js]: we don't need this, webpack do it itself

File: example/index.android.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+  render() {
+
+  const { twitter_access_token, twitter_token_secret, twitter_userId, error_message } = this.state
+
+    return (
+      <View style={styles.container}>
+      <TwitterLoginButton
+           accessible
+           accessibilityLabel={'loginButton'}
```
Comment (isnifer) [inline example/index.android.js]:
```jsx
accessibilityLabel="loginButton"
```

File: example/index.android.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+           onLoginFinished={this.onTwitterLoginFinished}
+           onLogoutFinished={() => alert("logout.")}/>
+        <Text
+           accessibilityLabel="twitter_response"
+           style={styles.instructions}>
+          { twitter_access_token != '' ? 'twitter_access_token: ' + twitter_access_token : ''} {'\n'}
```
Comment (isnifer) [inline example/index.android.js]: First of all:
1. `twitter_access_token` => `twitterAccessToken`
2. `!=` => `!==`
3. 
```js
'twitter_access_token: ' + twitter_access_token
``` 
=> 
```js 
`twitter_access_token: ${twitter_access_token}`
```

File: example/index.android.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+         })
+    } else if (result.isCancelled) {
+      alert("login is cancelled.");
+    } else {
+    console.log(result)
+          this.setState({
+            twitter_access_token: result.authToken.token,
+            twitter_token_secret: result.authToken.secret,
+            twitter_userId: result.userId,
+          })
```
Comment (isnifer) [inline example/index.android.js]: fix alignment

**Comment by fAns1k** [comment]
isnifer, could you do pull. Your screenshot seems like old version of sample.

File: src/TwitterLoginButton.android.js (hunk 4 above)
Comment (fAns1k) [inline src/TwitterLoginButton.android.js]: Is it necessary? It's just simple example.

File: src/index.js (hunk 5 above)
Comment (fAns1k) [inline src/index.js]: Are you sure? In sources of react native it used in every class.

File: src/TwitterLoginButton.android.js (hunk 4 above)
Comment (fAns1k) [inline src/TwitterLoginButton.android.js]: I think it does not matter 'cause this namespace is not visible for library users.

**Comment by itsmepetrov** [comment]
```
Appium is running on: 0.0.0.0:4723
Found next Android device: emulator-5554, version: 6.0

=======================================================================================
Selenium 2.0 / webdriver protocol bindings implementation with helper commands in nodejs.
For a complete list of commands, visit http://webdriver.io/api.html.
=======================================================================================


  Test sample auth with Twitter
    ✔  `Login with Twitter` button should be visible
---------------------------------------------------
SCREEENSHOT URL: http://i.imgur.com/ltRPj8S.png
---------------------------------------------------
---------------------------------------------------
Failed to save source: Invalid devkey
---------------------------------------------------
    ✔  User can click `Login with Twitter` button
    ✖  Error: element (//*/android.view.View[2]/android.widget.EditText[1]) still not visible after 60000ms at failure (/Users/anton/Documents/Projects/tipsi/tipsi-twitter/example/node_modules/tape-async/index.js:44:10)
        { [Error: element (//*/android.view.View[2]/android.widget.EditText[1]) still not visible after 60000ms] message: 'element (//*/android.view.View[2]/android.widget.EditText[1]) still not visible after 60000ms', type: 'WaitUntilTimeoutError' }undefined

passed: 2  failed: 1  of 3 tests  (1m 11.9s)

1 of 3 tests failed.
```

It looks like the `npm run ci` command does not run the `./configure` script before build.

@fAns1k please take a look

File: example/index.android.js [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+export default class example extends Component {
+  state = {
+    twitterUserId: '',
+    errorMessage: '',
+  }
+
+  handleCustomLoginPress = async () => {
+    try {
+      const result = await TwitterModule.logIn()
```
Comment (itsmepetrov) [inline example/index.android.js]: Please, rename `logIn` method to `login`

File: android/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<resources>
+    <string name="app_name">React native Twitter SDK</string>
```
Comment (isnifer) [inline android/src/main/res/values/strings.xml]: maybe just tipsi-twitter?

File: example/android/app/src/main/java/com/example/MainApplication.java [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+    private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
+        @Override
+        protected boolean getUseDeveloperSupport() {
+            return BuildConfig.DEBUG;
+        }
+
+        @Override
+        protected List<ReactPackage> getPackages() {
+            return Arrays.<ReactPackage>asList(
+                    new MainReactPackage(),
```
Comment (isnifer) [inline example/android/app/src/main/java/com/example/MainApplication.java]: maybe 4 spaces instead 8 ?

File: README.md [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+```xml
+    <application
+    ...
+        <meta-data
+            android:name="io.fabric.ApiKey"
+            android:value="YOUR_FABRIC_API_KEY" />
+    </application>
+```
+
+4. In your MainApplication.java add:
```
Comment (isnifer) [inline README.md]: This part should be removed from Readme

File: README.md [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+```
+
+
+## Usage
+
+```js
+...
+import { TwitterLoginButton } from 'tipsi-twitter'
+...
+
```
Comment (isnifer) [inline README.md]: Need to update example with TwitterModule.init method

File: README.md [hunk 14]
```diff
@@ -0,0 +1,173 @@
...
+    };
+...
+```
+
+
+## Usage
+
+```js
+...
+import { TwitterLoginButton } from 'tipsi-twitter'
```
Comment (isnifer) [inline README.md]: No more TwitterLoginButton here, just TwitterModule

**Comment by isnifer** [comment]
And one more time, I want to discuss the name of the module. I think we need to export the same module both on iOS and Android, for example:
```js
import { NativeModules } from 'react-native'

const { TPSTwitterModule } = NativeModules
export default TPSTwitterModule
```
This little trick will help us contain in `src` folder only two files - [TPSTwitterModule.js, index.js]

@fAns1k @itsmepetrov @cybergrind what do you think?

File: example/index.android.js (hunk 9 above)
Comment (fAns1k) [inline example/index.android.js]: Ok, but that's a moot point). https://ffeathers.wordpress.com/2014/04/06/login-or-log-in-a-way-to-choose-one-word-or-two/

File: example/android/app/src/main/java/com/example/MainApplication.java (hunk 11 above)
Comment (fAns1k) [inline example/android/app/src/main/java/com/example/MainApplication.java]: No, it's ok.

**Comment by itsmepetrov** [comment]
Everything is ok on both Android 5.1.1 and 6.0.0 emulators:

```
Appium is running on: 0.0.0.0:4723
Found next Android device: emulator-5554, version: 5.1.1

=======================================================================================
Selenium 2.0 / webdriver protocol bindings implementation with helper commands in nodejs.
For a complete list of commands, visit http://webdriver.io/api.html.
=======================================================================================


  Test sample auth with Twitter
    ✔  `Login with Twitter` button should be visible
    ✔  User can click `Login with Twitter` button
    ✔  loginInput should be visible
    ✔  User should be able to write login
    ✔  passwordInput should be visible
    ✔  User should be able to write password
    ✔  okButton should be visible
    ✔  User can click okButton
    ✔  User authenticated successfully

passed: 9  failed: 0  of 9 tests  (23.5s)

All of 9 tests passed!
```

**Comment by itsmepetrov** [comment]
@fAns1k please update Test section in README.md with information about `TWITTER_USER` and `TWITTER_PASS`

File: example/package.json [hunk 15]
```diff
@@ -0,0 +1,43 @@
+{
+	"name": "example",
+	"version": "0.0.1",
+	"private": true,
+	"scripts": {
+		"start": "node node_modules/react-native/local-cli/cli.js start",
+		"clean": "rm -rf android/app/release.keystore",
+		"appium": "appium",
+    "configure": "./configure.sh",
```
Comment (isnifer) [inline example/package.json]: fix indentation

File: example/android/app/src/main/AndroidManifest.xml [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+        android:configChanges="keyboard|keyboardHidden|orientation|screenSize">
+        <intent-filter>
+            <action android:name="android.intent.action.MAIN" />
+            <category android:name="android.intent.category.LAUNCHER" />
+        </intent-filter>
+      </activity>
+      <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />
+        <meta-data
+            android:name="io.fabric.ApiKey"
+            android:value="e81eb6c25b30623c26df84f4d77cfdeb220acc60" />
```
Comment (itsmepetrov) [inline example/android/app/src/main/AndroidManifest.xml]: We should use `./configure` script for that

**Comment by isnifer** [comment]
LGTM

**Pull request merged**